canary environment: Migrate from dbt to mz-deploy (#37123)
MaterializeInc/materialize: test/canary-environment/README.md

role trust configured for that connection's external ID before the S3-Tables
catalog will validate.

# Validating the workloads

The `test` workflow replaces the former `dbt test` data tests. It checks that:
- the RDS Postgres replication-slot count stays low (so Postgres does not run
  out of disk from stalled Materialize connections);
- every canary source is running (`mz_source_statuses`);
- every cluster replica is online (`mz_cluster_replica_statuses`);
- compute frontiers and their imports are within a minute of `mz_now()`;
- every monitored source-table and materialized view emits new rows on
  `SUBSCRIBE` (i.e. it is still making progress).

To run the validation:

1. Make sure the required environment variables are set correctly.
2. Run:

```
./mzcompose run test
```

# Developing the project

`mz-deploy compile` type-checks the whole project. It needs a `types.lock`
describing the columns of the `CREATE TABLE ... FROM SOURCE` tables, whose
schemas come from the live upstreams (RDS / Kafka Avro). `mz-deploy apply` (run
by the `create` workflow) and `mz-deploy lock` regenerate it, and it is
git-ignored for that reason. To compile offline without a connection, run

> `mz-deploy lock` once against the environment (or hand-write a `types.lock`).

Reviewer: ```suggestion
by the `create` workflow) and `mz-deploy lock` regenerate it.
```
